pre_commit_hooks/destroyed_symlinks.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import sys
+from operator import methodcaller
+from subprocess import check_call
+from subprocess import check_output
+from typing import Optional
+from typing import Sequence
+
+ORDINARY_CHANGED_ENTRIES_MARKER = b'1'
+PERMS_LINK = b'120000'
+PERMS_NONEXIST = b'000000'
+
+
+def normalize_content(content: bytes) -> bytes:
+    return b'\n'.join(
+        filter(
+            None,
+            map(
+                methodcaller('strip'),
+                content.splitlines(),
+            ),
+        ),
+    )
```

Comment:
don't use `map` / `filter` / `reduce` / `methodcaller` -- the equivalent list comprehension is 100% more readable

Reply:
what does this function do also? if it's to normalize newline endings it is not correct

Reply:
> what does this function do also? if it's to normalize newline endings it is not correct

Not exactly. Apart from normalizing newline(s) and removing them it should help to detect the problem in some weird but possible cases:
* ex-symlink was pointing to file with trailing whitespace(s) at the end
* ex-symlink was pointing to file with newlines and whitespaces before newlines

and the content of ex-symlink was modified by end-of-file-fixer and/or mixed-line-ending and/or trailing-whitespace hooks.
Yes, false-positives are possible because of that but they are extremely unlikely, much more unlikely than the weird symlinks themselves.

Reply:
do those scenarios actually happen?

Reply:
I don't think so. But at least added newline at the end is practically possible so, if we aren't going to support weird filenames, at least plain `.rstrip()` is required.